Adding test cases for <time> and <data> elements

arronei opened the pull request:
- Adding new test cases for &lt;time> and &lt;data> elements
- Added additional tests to &lt;output> test case

Commit:
Adding test case for the data element

Commit:
Adding more detailed tests for the output element

Commit:
Adding a new test for the data element

Commit:
Adding a fully detailed test case for the time element

zcorpan commented:
I think parser aspects are better tested in https://github.com/html5lib/html5lib-tests and imported to web-platform-tests with https://github.com/w3c/web-platform-tests/blob/master/html/tools/update_html5lib_tests.py

zcorpan commented on html/semantics/forms/the-output-element/output.html:
```diff
@@ -1,15 +1,31 @@
-<!DOCTYPE html>
+﻿<!DOCTYPE html>
```
What changed here? Byte order mark?

zcorpan commented on html/semantics/forms/the-output-element/output.html:
```diff
@@ -1,15 +1,31 @@
+  }, 'HTML parsing should locate 4 output elements in this document');
+  test(function () {
+    assert_true(!!window.HTMLOutputElement);
+  }, 'HTMLOutputElement should be exposed for prototyping');
```
This is already tested by https://github.com/w3c/web-platform-tests/blob/master/html/dom/interfaces.html so can be removed here.

zcorpan commented on html/semantics/forms/the-output-element/output.html:
```diff
@@ -1,15 +1,31 @@
+  test(function () {
+    assert_true(document.createElement('output') instanceof window.HTMLOutputElement, 'createElement variant');
+    assert_true(outputs[0] instanceof window.HTMLOutputElement, 'HTML parsing variant');
+  }, 'the output elements should be instanceof HTMLOutputElement');
```
This also -- well technically not "HTML parsing variant" but no browser fails that so it's not an interesting test in addition to `createElement` IMHO :-)

SimonSapin committed:
Add tests for URL setters

A lot of them fail in browsers, but I believe they’re should pass
in a spec-conforming implementation.
(With a few exceptions noted in comments.)

SimonSapin committed:
Revert URL setter test spec deviations for now

See https://github.com/w3c/web-platform-tests/pull/2830#issuecomment-210698473.

This can be un-reverted when the following are fixed in the spec:

- https://github.com/whatwg/url/pull/113
- https://github.com/whatwg/url/issues/104

zcorpan committed:
change/input events on radio should be trusted

This reverts
5d1a5c182cf0cddb4690613f913c8abd427946cf

Also test that isTrusted is false for the click event.

Ref.
https://github.com/whatwg/html/issues/1126
https://github.com/w3c/web-platform-tests/issues/1418#issuecomment-214615790
https://github.com/w3c/web-platform-tests/pull/2991

Commit:
Fixed bug: private key usage possibilities

domenic committed:
Fix createEvent("popstateevent") test; it should work (#2964); r=Ms2ger

See https://dom.spec.whatwg.org/#dom-document-createevent, which changed in https://github.com/whatwg/dom/commit/9e3ce67c7927d6642646a3d0c84fa6d8f7926cfa.

Commit:
Don't remove the body element in tests, and test setting color attributes with no body (#2951)

plehegar committed:
missing js dependency

kojiishi committed:
Move pre-v1 custom elements tests to "v0" directory for shorter path (#2965)

Ms2ger committed:
Remove myself from the domparsing owners.

Commit:
Rename the ReferrerPolicy "origin-only" enum value to "origin";

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265941

Commit:
Update request-init-001.sub.html as well

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265941

bzbarsky committed:
When throwing an exception from XHR open() due to not being in an active document, make sure it's an InvalidStateError.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265970

Commit:
Update request-init-003.sub.html as well

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265941

birtles committed:
Move getComputedTiming-* tests to timing-model;

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1266257

birtles committed:
Split iteration progress tests into 'simple iteration progress' and 'active time' tests;

MozReview-Commit-ID: 1fc6Wpi9nN8

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1266257

Commit:
Throw InvalidStateError at IDBCursor.update() if IDBCursor.continue() is in progress.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265973

Commit:
W3C Reflect referrerPolicy as a limited enumerated attribute. r=franziskuskiefer .

MozReview-Commit-ID: Dppj4iuZnFp

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1260664

hashedhyphen committed:
Extract useful keyframes tests to a new file

Extract a lot of tests from keyframe-effect/constructor.html and add some
documentation so that we can reuse them in other tests. This helper will
be used when we want to test these API:

* KeyframeEffect(ReadOnly) Constructor
* KeyframeEffect.setFrames()
* SharedKeyframeList Constructor
* Animatable.animate()

and so on.

MozReview-Commit-ID: 5jZbXyPFHih

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1244591

hashedhyphen committed:
Add web-platform tests for KeyframeEffect.setFrames

Some tests are expected to fail until Bug 1216843 and 1216844 are resolved, so that
we specify that on the meta file `setFrames.html.ini`.

MozReview-Commit-ID: 6z2P1KkGJhb

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1244591

wanderview committed:
P2 Add a web-platform-test for the window navigation case.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265795

bzbarsky committed:
Fix <base> href getter to follow the spec; it should be using the fallback base URI to resolve against, not the document URI.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1266077

jdm committed:
Accept opaqueredirection fetch results if the request redirection type is manual.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1209095

wanderview committed:
P2 Expand navigate-window.https.html wpt test to cover uncontrolled windows.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265771

jgraham committed:
Add some tests stolen from https://github.com/w3c/web-platform-tests/pull/493 so I (<bzbarsky@mit.edu>) can land test coverage together with the code change.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1266831

bzbarsky committed:
Make <xmp> and <listing> use HTMLPreElement as their primary interface, per <https://github.com/whatwg/html/issues/1015>.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1266851

Commit:
text-decoration-color: currentcolor should not use value from -webkit-text-fill-color;

MozReview-Commit-ID: 2SfENZieSzQ

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1266948

birtles committed:
Add assert_times_equal for comparing times while allowing for a degree of imprecision;

MozReview-Commit-ID: LAuCLmchBHa

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267893

birtles committed:
Move existing timing model tests into animation-effects subfolder;

MozReview-Commit-ID: JojGDs8aWAp

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267893

birtles committed:
Add tests for start time;

MozReview-Commit-ID: 7yhUVSEIR2v

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267893

birtles committed:
Follow-up to fix mistaken precision value in changeset c2f257b08990;

The precision value was for seconds despite the fact that we are comparing
milliseconds. This was causing intermittent failures such as:

  https://treeherder.mozilla.org/logviewer.html#?job_id=26711451&repo=mozilla-inbound

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267893

Commit:
Update web platform tests.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1206616

wanderview committed:
P4 Add a wpt test that verifies a service worker update can recover from a broken navigation interception.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267733

Commit:
Update web platform tests (r=bkelly)

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1206961

hashedhyphen committed:
Make tests to use the common test data in keyframe-utils.js

* Add new tests for single-valued keyframe to gKeyframeSequenceTests.
* Update constructor.html.ini and setFrames.html.ini so that we can use the new tests.
* Iterate all the possible tests for keyframes in animate.html (and modify .html.ini).
* Use `null` for some keyframes because assert_class_string doesn't care about details.
* Remove unnecessary call `done();`.

MozReview-Commit-ID: 6209DO9BYFP

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265274

emilio committed:
Implement DOMTokenlist.replace r=baku,Ms2ger

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1224186

BorisChiou committed:
Test for KeyframeEffectReadOnly with null target.

MozReview-Commit-ID: DQ5k6W1bgUC

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1067769

BorisChiou committed:
Add some simple tests for document.getAnimation() in wpt.

MozReview-Commit-ID: Ip2XYKMfUjb

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1067769

BorisChiou committed:
Test for setting the target in basic cases.

MozReview-Commit-ID: HG1Ttqzkti7

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1067769

bzbarsky committed:
Remove a race in the picture-tag cases in mixed-content web platform tests.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1184351

bzbarsky committed:
Fix mixed-content web platform tests to not screw up their bytes on Windows.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268915

bzbarsky committed:
The load events we fire on iframe/frame/object (in the document case) should not be cancelable.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268953

jwatt committed:
Add Web Platform Tests for Secure Contexts.

MozReview-Commit-ID: JU2bzytnkkI

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1162772

Commit:
W3C referrerpolicy content attribute doesn't work in FF. r=josh,

MozReview-Commit-ID: KKq7ZRJziBu

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1261003

jswalden committed:
Permit a cyclic [[Prototype]] chain to be created through a Location object.  r=bz,

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=888969

EdgarChen committed:
Add more web-platform-tests for relevant-mutations;

MozReview-Commit-ID: FCeDkPE3smy

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1193349

bzbarsky committed:
Fix the meta-csp mixed-content web platform tests to actually output the <meta> tag with the content policy, so they're testing something useful.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1270219

bzbarsky committed:
Start allowing sync XHR with credentials in window scopes again, since no one except for us bothered to implement the spec.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1260515

bzbarsky committed:
Implement DOMTokenList.prototype.supports().

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1257849

Commit:
Allow objectStores and indexes to be renamed.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1118028

wanderview committed:
Fix wpt stream tests to reference new service-worker test-helpers.sub.js.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1270673

ayg committed:
track kind=invalid> should behave like metadata, not subtitles;

This follows a spec change <https://github.com/whatwg/html/issues/293>,
which AFAIK no other browser has implemented, so it has some regression
potential.

The web-platform tests changed are out-of-date and match the old spec,
so I'm changing them here to match the new spec.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1269712

dbaron committed:
patch 4 - Support #rgba and #rrggbbaa colors in CSS.

This adds support for #rgba and #rrggbbaa colors to CSS.  This feature
is specified in https://drafts.csswg.org/css-color-4/#hex-notation .

This adds new types to nsCSSValue so that we can serialize the syntax
that was specified, as we do for other distinctions in how colors are
specified.

It does not change the behavior of the hashless color quirk, which
continues to support only 3 and 6 digit colors as specified in
https://quirks.spec.whatwg.org/#the-hashless-hex-color-quirk (step 4).

This changes property_database.js to remove various uses of 4 and 8
digit colors as invalid values.  It then adds them in slightly fewer
places as valid values, but more thoroughly testing both initial and
non-initial values on 'color'.

It marks two canvas tests explicitly testing this feature as no longer
known to fail by removing their .ini files.

Finally, it adjusts the web platform test testing the hashless color
quirk to no longer treat 4 and 8 digit colors with hashes as invalid
values.  Removing the relevant test items seems like the right thing
since they're in a section where 3 and 6 digit colors were skipped but
other lengths were tested.  Modifying this imported test is OK since:
  <jgraham> dbaron: Commit the change you want to m-c, it is
    (semi-)automatically upstreamed every so often (typically
    about once a week)

MozReview-Commit-ID: IFq4HxaRkil

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=567283

jgraham committed:
Fix line endings causing lint errors.

plehegar committed:
Use step_timeout (#2966)

* basic timeout cleanup

* cross timeout cleanup

Ms2ger committed:
Merge the two DOM Parsing directories.

mrmiywj committed:
'closeevent' argument for Document::createEvent

fix alphabetical order

delete unused tests

domenic committed:
Add streams test owners

domenic committed:
Add @wanderview to streams test owners as well

domenic committed:
Fix service worker helper reference in two streams test files

74267b2e2c06a6a9445f21d680194161ddacd3f3 missed a couple files.

Commit:
Add a test to check window.frames doesn't leak Shadow DOM. (#3019)

Ms2ger committed:
Move init-event-while-dispatching.html to DOM.

This is a valuable test for all types of events, not just those defined in
uievents.

Ms2ger committed:
Move event-phases-order.html to DOM.

Also stop depending on the click() method.

Ms2ger committed:
Move constructors.html to DOM.

Ms2ger committed:
Move ClickFakeEvent.nondocument.html to DOM.

Ms2ger committed:
Merge two uievents tests for exceptions in event listeners and move them to DOM.

Commit:
Removing interactive control of tests. Test harness can't work with it. Tests very slow again, due to always running RSA.

alancutter committed:
Remove duplicate test case (#3022)

zcorpan committed:
Fix #2757: Test input/textarea.selection{Direction,Start,End} not throwing (#2825)

This tests https://github.com/whatwg/html/pull/1006.

Commit:
Workers working, though usually timing out

birtles committed:
Add test guidelines to web-animations/README.md

birtles committed:
Hopefully fix trailing whitespace lint error

Commit:
Add tests checking same promise and battery manager are created per Navigator object (#2962)

Corresponding spec changes are at https://github.com/w3c/battery/pull/3
and the original issue is at https://github.com/w3c/battery/issues/2

Ms2ger committed:
Move DOM.event.flow.html to DOM.

Ms2ger committed:
Rewrite Event-dispatch-bubbles-true.html.

Ms2ger committed:
Move EventListener.eventHandler.html to DOM.

Ms2ger committed:
Refactor EventListener-handleEvent.html.

Ms2ger committed:
Move EventObject.multiple.dispatchEvent.html to DOM.

Ms2ger committed:
Refactor Event-dispatch-multiple-stopPropagation.html.

wseltzer committed:
Create OWNERS

zcorpan committed:
Fix #2924: Update tests for <label form> removal spec change (#2926)

This tests https://github.com/whatwg/html/pull/1120.

Commit:
Cleaned up dead code

jimsch committed:
Add an IDL test set to the suite based on current document

Note - I needed to comment out two lines. I don't know if this is an
error in the IDL in the document or something else.

jimsch committed:
Fix the lint bug

jimsch committed:
Don't include windows

Commit:
getElement* API should not leak nodes in shadow tree. (#2922)

kojiishi committed:
Custom Elements: Element definition (#2940)

* Custom Elements: Element definition

* domenic review

* domenic review 2

* Fix ohter "new NotSupportedError" to "NOT_SUPPORTED_ERR"

* Changed dup-name/dup-ctor tests to one test

* dominiccooney review

Ms2ger committed:
Move the link element to the right place. (#3029)

Thanks to zcorpan for noticing.

zcorpan committed:
Change .headers and .ping to be DOMString/USVString (#3028)

Update for https://github.com/whatwg/html/pull/1262.

jimsch committed:
Add workers to the test

jimsch committed:
Remove comments on interface

birtles committed:
Move keyframe sequence test data to appropriate place in keyframe-utils.js;

MozReview-Commit-ID: 8laNGaElWz6

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268858

bzbarsky committed:
Some javascript:/navigation web platform test fixups.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268047

bzbarsky committed:
Change javascript: evaluation to ignore the return value if it's not a string.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268047

chihweitung committed:
P2 tests for response.redirected with and without cached. r=bkelly.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1243792

Commit:
Add Test Coverage in Web Platform Tests.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268380

chihweitung committed:
P1 implement response.redirected - Fetch API & Cache API & ServiceWorker changed. r=bkelly, r=baku.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1243792

bzbarsky committed:
Add the sandbox propagates to auxiliary browsing contexts flag to iframe sandboxing.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1190641

chihweitung committed:
caches.match with wrong name should resolve to undefined.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1271069

birtles committed:
Rename KeyframeEffectReadOnly.getFrames() and KeyframeEffect.setFrames() to getKeyframes()/setKeyframes(); r=hiro,

MozReview-Commit-ID: GwLLY39l1KE

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1271904

birtles committed:
Add tests that getKeyframes() returns the correct result when CSS variables are used;

MozReview-Commit-ID: BZrd3FAhrrf

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1268858

jgraham committed:
Fix too-short timeouts in CSP tests, rs=bz

MozReview-Commit-ID: 6O6lal4a2aM

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1273093

birtles committed:
Move web-animations/animation-model/keyframes to keyframe-effects;

To better match the outline of the spec.

MozReview-Commit-ID: LZrDJvql65s

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264822

jgraham committed:
Make test that depends on focus wait for the window to have focus, rs=bz

MozReview-Commit-ID: HvvQQXWIQcO

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1271877

birtles committed:
Move web-animations/keyframe-effect/keyframe-handling.html;

To better match where the behavior under test is defined in the spec.

MozReview-Commit-ID: 6Uhypq7ZxDc

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264822

birtles committed:
Add a test that non-animatable properties are not accessed;

We already have a test under
web-animations/animation-model/animation-types/not-animatable.html that checks
that non-animatable properties don't show up in the result from getKeyframes().
However, the spec says we should not even *read* those properties off the input
object(s).

This patch adds a test to check that we don't read non-animatable properties.
In a later patch we will remove the existing test since it will be redundant
after this test is added.

MozReview-Commit-ID: J11cUybTLcX

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264822

birtles committed:
Ignore shorthand properties when all their subproperties are not-animatable;

This patch also includes a tweak to not-animatable.html to match the order in
which properties are enumerated on the object. This test was always in error but
we never noticed since the test failed before reaching the test in question.

Making the test dependent on the order in which object properties is enumerated
is not good but we will remove this test in the next patch. In this patch we
just make sure it passes.

MozReview-Commit-ID: AKXdHj4nUMv

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264822

birtles committed:
Remove web-animations/animation-model/animation-types/not-animatable.html;

Part 3 in this patch series introduces a test that checks this behavior more
strictly. If that test passes, this test is sure to pass as well.

MozReview-Commit-ID: 8UvESeSYYFl

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264822

upsuper committed:
Make fullscreen enabled flag not be affected after document is loaded.

MozReview-Commit-ID: L2dMAUr63qv

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1270648

upsuper committed:
Setup the sandboxed fullscreen flag and make it work properly.

MozReview-Commit-ID: 3EKRbOyXwNi

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1270648

Commit:
Modify the Test Case to Run the Test on |WorkerGlobalScope| instead of |WorkerUtils|.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1272551

johndai1984 committed:
Update web-platform tests for reportValidity().
From eb8351d4f84cb69cff2bc431539158ad046d51ee Mon Sep 17 00:00:00 2001

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1088761

johndai1984 committed:
HTMLTableElement.rows return element in the order.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264947

mantaroh committed:
Part1. Rename web-animations test folders in web-platform-tests to align with the spec.

MozReview-Commit-ID: GeOwu2FhJnH

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264516

mantaroh committed:
Part2. Move api tests of animation's into interfaces directory.

MozReview-Commit-ID: CASNLVJdzhT

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1264516

bzbarsky committed:
Add a test for EventListener incumbent globals.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1273529

Commit:
enable pseudo class :dir wpt test cases.

MozReview-Commit-ID: Hlc9KOoKWRX

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1270713

jgraham committed:
Remove bogus usage of console.log

youennf committed:
Fix fetch test in case of manual redirection mode and bad location URL (#3057)

rniwa committed:
Add a test for scrolling to a fragment inside a shadow tree. (#2953)

As discussed on https://github.com/w3c/webcomponents/issues/66 and https://github.com/whatwg/html/issues/1192,
the user agent should not use an element that has an ID exactly equal to decoded fragid or an anchor element
with a name attribute exactly equal to fragid if the element inside a shadow tree as the indicated part of the document.

WebKit nightly builds and Google Chrome Canary both passes the test.

rniwa committed:
Add a test for HTMLSlotElement interface (#2929)

Import WebKit's test for HTMLSlotElement interface's test attribute and assignedNodes() from
http://trac.webkit.org/browser/trunk/LayoutTests/fast/shadow-dom/HTMLSlotElement-interface.html?rev=200285

All test cases pass on WebKit nightly builds, and some test cases fail on Google Chrome Canary builds
because the latter doesn't clear the list of assigned nodes when a slot element is removed from a tree.

Commit:
Fixes nits in PR#2929 and PR#2953 (#3058)

youennf committed:
Adding tests for Response redirected attribute (#3061)

Ms2ger committed:
Remove reference to urltestparser.js (fixes #3065).

The file was removed in 97342db760bc0e00395750b9322e24d67480ea33.

jgraham committed:
Update to latest testharness.js

EdgarChen committed:
Revert 3898ec "Fix <img sizes='calc(1px'> test"

Fixes #3066.
PR: #3069

andreastt committed:
webdriver: remove timeout debug statement

domenic committed:
Remove invalid-as-of-ES6 test of ProgressEvent.length (#3068)

As of ES6, the length property on functions is configurable, and can be deleted.

Commit:
Reduced RSA test cases to run in a reasonable time

zcorpan committed:
Test label.form when its control is a progress or meter element (#3071)

Ref. https://github.com/whatwg/html/issues/1308

hallvors committed:
Ensuring URLs in XHR invalid URL test are invalid (#3072)

Ref. https://bugzilla.mozilla.org/show_bug.cgi?id=918735

Commit:
Convert deepPath() to composedPath(). (#3076)

Event.deepPath() was renamed to composedPath().
See
https://github.com/whatwg/dom/issues/242

rianby64 committed:
Fix some <body onerror>/window.onerror tests (#3079)

The fourth arg's name changed from "column" to "colno" in the spec
https://github.com/whatwg/html/commit/9f5fd3e4fb31ea3b168dbeca1624fb604bf7479b
https://html.spec.whatwg.org/multipage/webappapis.html#getting-the-current-value-of-the-event-handler

Also fix *-error-data-url.html tests to avoid testing serialization of a data: URL with a space.

Ms2ger committed:
Remove the one remaining test for EventException. (#3078)

This case is tested in dom/events/EventTarget-dispatchEvent.html.

Commit:
First working test of encrypt (with AES-GCM)

Commit:
Fixed too early return from buffer comparisons.

Commit:
Adding more test data.

jimsch committed:
Change the owners list

Add myself and charles to the owner list.

zcorpan committed:
Add scheme-only URL tests (#3073)

Ref. https://bugzilla.mozilla.org/show_bug.cgi?id=918735

cbrewster committed:
Added tests for formsubmitter methods

KiChjang committed:
Add test for encoding text/plain in forms

Commit:
Avoid panics when using HTMLAnchorElement attribute setters

Fixes #10877.
Includes new test for attribute getters and setters.

emilio committed:
codegen: Throw on an invalid enum value when appropiate

KiChjang committed:
Fire 'select' event in SetSelectionRange

mrmiywj committed:
htmlanchorelement getter work before href setting is called

update tests

rewrite it

Manishearth committed:
Add tests for submit buttons with values

rebstar6 committed:
Fix the tests for origin-when-cross-origin

It's origin-when-cross-origin, not origin-when-crossorigin.

Ms2ger committed:
Rewrite customevent.html as a testharness.js test. (#3096)

annevk committed:
Redirects to non-HTTP(S) schemes are a network error

See https://github.com/whatwg/fetch/issues/309 for context.

Commit:
Remove irrelevant/obsolete event tests in shadow-dom. (#3097)

Introduction of "composed" flag in Event reverts the semantics of
"scoped", and the default is now changed, so some of tests do not
make sense.  Removing such tests.

kojiishi committed:
Fix issues in custom-elements/custom-elements-registry/define.html

* Names should not use uppercase, or it will throw a SyntaxError.
* 8. observedAttributes rethrow test should use a static getter.
* 12-17. Add more types to test IsCallable for callbacks.

Commit:
Cleaned up first test. Now ready to add remaining AES-GCM tests.

Commit:
All GCM tests with 128 bit tags work

NavidZ committed:
Change the expected width/height value of PE

Changing the expected value of width/height
properties of pointer events to 1.

Commit:
All GCM successes tested

Commit:
Added failing tests.

naeloued committed:
Add new tests for fetch request on blob resource. (#3094)

* Add new tests for fetch request on blob resource.

* Fix: method is passed as a String instead of Array

Commit:
Added a webdriver test for resizing the window.

Commit:
AES-GCM testing complete

andreastt committed:
Lint webdriver resize (#3102)

* webdriver: lint window resize test

No functional changes.

* webdriver: rename resize.py to contexts.py

We ought to name files after the corresponding chapters in the
specification.

Commit:
Added AES-CBC tests

Commit:
Cleaned up AES-GCM tests

Commit:
Fixed worker imports

Commit:
Worker and timeouts fixed

Commit:
Minor cleanup for vector definitions

Commit:
Added AES-CTR tests

alvestrand committed:
Fixed constraint syntax

Some of the tests were using either old syntax or a mixture of old and
new syntax for constraints. These should now be correct.

Commit:
Cleaning up and consolidating AES tests

Commit:
Remaining consolidation of AES tests.

NavidZ committed:
Update resource submodule to the latest version

travisleithead committed:
Testing the HTML entities in XHTML with various Doctype defs (#3081)

Commit:
First RSA test works

Commit:
Testing by roundtrip, and verifying that second encrypt gives different result from first one

Commit:
Plaintext should be too long for bigger hash functions. WIP to fix tests.

Commit:
RSA tests complete.

Commit:
Renamed directory to include decrypt tests, too.

Commit:
RSA-OAEP encrypt and decrypt tested

Commit:
Added failure tests to RSA-OAEP

Commit:
Comprehensive decrypt tests in place.

NavidZ committed:
Add a test for pointer event clientX and clientY

sideshowbarker committed:
cc Update tests to reflect style[scoped] removal

Honry committed:
Fix bug in the battery-promise.html (#3108)

Fix issue https://github.com/w3c/web-platform-tests/issues/3107

Commit:
Added CBC padding error tests

rebstar6 committed:
Change crossorigin to cross-origin in json, regenerate (#3115)

NavidZ committed:
Use bounding rect

Ms2ger committed:
Remove Node::rootNode. (#3125)

See <https://github.com/whatwg/dom/commit/42a45eb0d1af13e510620b6b29696997f0c7aa3d>.

alvestrand committed:
Rename getUserMedia-using tests to .https.html

This should make it possible for tools to detect that they need
to be served over https.

rebstar6 committed:
Change http-csp delivery to referrer policy header (#3118)

Ms2ger committed:
Fix 'Unscopeable' typos in interfaces.html. (#3128)

Commit:
Added AES-GCM without additional data element tests

Commit:
Properly detect test completion and call done() method.

Commit:
Properly detect test completion for RSA tests, too, and call done() method.

Commit:
Removed step_timeout to detect completion.

Commit:
Checked that data cloning took place for AES

Commit:
Added tests for altered buffers

Commit:
Added test for mismatched algorithm and key

NavidZ committed:
Change assert_between_inclusive to manual boundary test

r12a committed:
encoding legacy directories (#3064)

The overwhelming majority of the cases are tabs for the somewhat more complex than usual javascript that drives the tests. I wanted to retain those so that i can read the code easily. I've run all the tests sufficiently often already to believe that these tabs aren't affecting the results.

Ms2ger committed:
Update tools.

Ms2ger committed:
Update tools.

Ms2ger committed:
Update tools.

RByers committed:
Add test for AddEventListenerOptions passive member (#3119)

Also includes a minor style cleanup to the related capture option test.

taff-franck committed:
Basic tests for Presentation API (#3063)

* PresentationRequest constructor  test with incorrect url
* PresentationRequest test with correct url
* reconnect() test with wrong Presentation ID

yhe39 committed:
Add 3 presentation connection lifecycle test cases (#3114)

mariuswsl committed:
Basic Use Case tests for the Presentation API (#3062)

Tests for the Use Cases 'monitor availability of presentation displays', 'starting a new presentation' and 'reconnect to a presentation' for the Presentation API

Commit:
Digest tests first working commit

Commit:
Digest method tests complete

bzbarsky committed:
Add another test for incumbent globals in addEventListener.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1273932

bzbarsky committed:
Fix the handling of numeric entities in xml-stylesheet href values.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1273282

bzbarsky committed:
Remove the HTML Microdata API, since no one else ended up implementing it and now it's been removed from the spec.  r=bkelly,jgraham

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=909633

bzbarsky committed:
Fix fallback loading of <embed> inside an <object> with no type attribute to actually work reliably.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1272651

bzbarsky committed:
Make suppressing event handling on a document also block script execution for that document.

Note that we do this here instead of when we suspend timeouts, because it makes
it simpler to handle situations in which a window's document changes while
things are suspended/blocked.

The nsDocumentViewer.cpp is not strictly needed, but avoids some extraneous
calls to SuppressEventHandling with a 0 suppression count.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1267989

mantaroh committed:
Add animation mochitest for examine timeline is null.

MozReview-Commit-ID: 3QFFW0qZdyO

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1096776

mantaroh committed:
Modify animation in order to support null timeline.

MozReview-Commit-ID: 5iKUtNuqKeX

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1096776

wanderview committed:
P2 Fix register-wait-forever-in-install-worker.https.html to expect new unified service worker job queue behavior.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1273920

chihweitung committed:
Intermittent fetch-request-redirect.https.html - test modified.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1272830

chihweitung committed:
Reformat for fetch-request-redirect.https.html & fetch-rewrite-worker.js.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1272830

Commit:
change wpt test origin-when-crossorigin to origin-when-cross-origin.

MozReview-Commit-ID: DProAfnX6ai

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1270050

hopanghsu committed:
Add a testcase for Node.isConnected.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1261002

daliacoss committed:
Fix incorrect test assertion in 'getResponse: don't expose x-nonexposed' and 'Combined testing of cors response headers'.

MozReview-Commit-ID: HD0TBJcnxst

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1261472

birtles committed:
Add tests for setting the timeline of an animation

MozReview-Commit-ID: 1xdKJONyFJV

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1178662

dadaa committed:
Modified and append tests.

MozReview-Commit-ID: A9gGXe1vMal

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1272211

Commit:
Do not use Math.random() to create unique values to avoid intermittent failrues.

MozReview-Commit-ID: 3QHrpCTpEzM

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1276490

vyv03354 committed:
Open binary files in binary mode.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1278381

dminor committed:
Add webplatform-test for IIRFilterNode;

MozReview-Commit-ID: qSDxvk60j2

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1265408

jgraham committed:
Fix testharness.js path in subresources-integrity tests,

MozReview-Commit-ID: GXpCHvZ427l

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1278694

asutherland committed:
P1 Fetch "only-if-cached" cache mode tests.

request-error.html:
- Add test verifying we enforce only-if-cached requires same-origin mode.

request-cache.html (only-if-cached does not hit network):
- Add comment block explaining the test dictionary arguments.
- Eliminate duplicated code in populate_cache which seems to have existed only
  for historical reasons and is now moot.
- Add `response` optional test dictionary array for cases where a network error
  is expected as the result of the fetch.  Normalize server results to handle
  this case since test cases may no longer talk to the server at all.
- Add only-if-cached tests that verify the network is not touched in any of
  the fresh cached, stale cached, and not cached cases.

request-cache.html (cross-origin redirects in the cache are not traversed):
- Tests added to request/request-cache.html instead of under redirect/ because
  redirect/ and redirect.py explicitly disable caching.  Also, the issue is
  about cache behavior, not the redirect-following logic.
- Enhance cache.py to accept a "redirect" parameter that generates a 302
  redirect using a Location header when received.
- Make the "error" handling generate a test failure if we don't fail when
  expected.
- Add "redirect" argument to request-cache.html test info dicts to tell
  cache.py to generate a redirect.
- Add test cases for "only-if-cached" to first prime with redirect, expecting
  success on same-origin redirects and failure on cross-origin redirects.

Upstreamed from https://bugzilla.mozilla.org/show_bug.cgi?id=1272436

Commit:
Fixed position of buffer change. Removed using Promise.all for done().

jgraham committed:
Update to latest wpt-tools

tidoust committed:
Update IDL definitions to match current spec (#3157)

This commit updates the IDL definitions that get tested. Main changes:

1. Update the signature of appendStream to use ReadableStream
2. Add the trackDefaults attribute to SourceBuffer
3. Add TrackDefaultList and TrackDefault interfaces
4. Add event handler attributes
5. kinds attribute replaced by getKinds method

nox committed:
Implement [Unscopable] (fixes #11583)

kevgs committed:
Make Document::DefaultView return a null value when there's no browsing context

achals committed:
Implement URL.domainToUnicode

Commit:
Added a test for changing an iframe's parentage.

Commit:
Fixed time of altering source buffer test

Commit:
Improved test labels for bad tag lengths

Commit:
Fixed test data for some tests that should fail, but need legal data inputs.

Commit:
Added test for whether only fully active documents contribute to the joint session history.

Manishearth committed:
Make send-usp tests less timeout-prone

Commit:
Added tests for cross-origin iframe parentage.

jgraham committed:
Update to latest wpt-tools

Commit:
Battery: add a test loading an iframe page that is NOT about:blank (#3167)

Also split the tests checking a battery promise per navigator because
onmessage is used in both iframe and new window scenarios.

Resolve #3107.

Commit:
Import test cases from DVCS for Device Orientation Event (#350)

* Import test cases from DVCS for Device Orientation Event

https://dvcs.w3.org/hg/geo/rev/5ee2a7297634

* Update test cases for DeviceOrientation Event

- Turn device_orientation_api.html and manifest.txt files into mainifest.txt and TODO.txt
- Remove t026.html as it has same checkpoints as screen-upmost.html, and merge values-null.html into screen-upmost.html
- Remove t030.html, t031.html, t032.html as they are incorrect in null attributes checking
- Rewrite other test cases

* Remove http://w3c-test.org from test files.

* Address @darobin comments:

- Rename the manual tests as -manual
- Drop the manifest.txt
- Remove all usage of user-scalable=no

* Use WebIDLParser.js

* Address @Honry comments

- Indent consistently using spaces
- Put user interaction instructions into a paragraph
- Remove unreferenced variable

* Remove trailing whitespace

* Remove TODO as tests shall follow the latest spec at

https://w3c.github.io/deviceorientation/spec-source-orientation.html

renchenglei committed:
Sync WebGL conformance tests to lastest (#3122)

siusin committed:
updated the interfaces of PointerLock (#3171)

* updated the interfaces of PointerLock

* updated the idl of PointerLock

ddorwin committed:
Add Blink layout tests

Based on Chromium commit 8c4625186dffe3eae184780a1f9690eb80ef0074.

Commit:
Merge branch 'master' of https://github.com/arronei/web-platform-tests

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

zcorpan commented:
In the future, can you do a rebase instead? (Or not; no reason if there's no conflicts...)
